### DISCUSSION TITLE
Who has experience converting OpenCasCade TopoDS_Shape data structures to ASSIMP scene?

### DISCUSSION THREAD

**Discussion by wujie-zxc** [assimp/assimp#3630] [Q&A]
void assimp_export(std::vector<**TopoDS_Shape***> &vec, const char *file)
{
    Assimp::Exporter exporter;
    aiScene t_scene;
    //I want to parse the OCC data to fill the scene , then export scene

}

**Comment by kimkulling** [comment]
Do you have a documentation about the internals of the data-structure for a TopoDS_Shape? Then we can tr to help you here.

**Reply by PaddiM** [reply]
If you have a TopoDS_Shape you can triangulate it and use the triangulation to fill an aiMesh. You can use the STL writer from OpenCasCade as a sample.

**Comment by sharjith** [comment]
With a lot of help from ChatGPT, I have achieved this to some extent. Check out my code here:
https://github.com/sharjith/ModelViewer-Qt
